subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/ArtifactTransformDependencies.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts.transform;
+
+import org.gradle.api.Incubating;
+
+import java.io.File;
+import java.util.Collections;
+
+/**
+ * An injectable service that when injected into a {@link ArtifactTransform} can be used to access
+ * the dependency artifacts of the artifact being transformed.
+ *
+ * @since 5.1
+ */
+@Incubating
+public interface ArtifactTransformDependencies {
+    /**
+     * Returns the dependency artifacts of the artifact being transformed.
+     * The order of the files match that of the dependencies in the source artifact view.
+     */
+    Iterable<File> getDependencies();
```

Comment:
:x: can we call this `getArtifacts()`? It may be that we need to provide "real" dependencies in the future, with more metadata.

Reply:
I'd rather call this `getFiles()`, since we are returning files.